src/dialects/postgres/query-interface.js

```diff
@@ -160,7 +160,7 @@ export class PostgresQueryInterface extends QueryInterface {
     const query = this.queryGenerator.getForeignKeyReferencesQuery(table.tableName || table, this.sequelize.config.database);
     const result = await this.sequelize.queryRaw(query, queryOptions);
 
-    return result.map(Utils.camelizeObjectKeys);
+    return result.map(Utils.camelizeKeys);
```

Comment:
What is the reason behind this change? I would expect this method to continue returning an array of objects, so the previous code should be enough

Reply:
The new function is recursive to traverse the argument.  `camelizeObjectKeys` doesn't camelize nested objects (or a nested array of objects).  

It handles a simple array of objects well using `.map(camelizeObjectKeys)`, but for anything more complex, there are gaps.  `camelizeObjectKeys` is also more efficient because it doesn't have the additional function calls, which is why it deserves to live on.

This:
```json
[
  [
    {
      "table_name": "stores",
      "table_column_names": ["customer_name", "customer_phone"],
      "references_table_name": "customers",
      "references_table_column_names": ["customer_name", "customer_phone"]
    },
    {
      "table_name": "cars",
      "table_column_names": ["part_id"],
      "references_table_name": "parts",
      "references_table_column_names": ["id"]
    }
  ],
  {
    "meta": {

    }
  }
]
 ```
Becomes:
```json
[
  [
    {
      "tableNames": "stores",
      "tableColumnNames": ["customer_name", "customer_phone"],
      "referencesTable": "customers",
      "referencesTableColumnNames": ["customer_name", "customer_phone"]
    },
    {
      "tableNames": "cars",
      "tableColumnNames": ["part_id"],
      "referencesTable": "parts",
      "referencesTableColumnNames": ["id"]
    }
  ],
  {
    "meta": {

    }
  }
]
 ```
 
I initially intended to camelize the values too, but I changed my mind as I thought this information should be raw; for instance, if building an ERD, you want to see the actual values stored in the DB.  If working through the ORM, you want to work with camelized values and that should happen elsewhere.

Reply:
> I initially intended to camelize the values too

Yes they definitely should not be camelized.

Still, why is this needed here? This method should return a flat array of flat objects. There is no need for a deep call here?

Reply:
I just ran this code and it does return a flat array of flat objects. There is no need to make this recursive

![image](https://user-images.githubusercontent.com/1280915/172075978-1815e6c1-8ea8-4fc2-ab03-6d1f4f61d2ec.png)